jax/experimental/host_callback.py

```diff
@@ -11,12 +11,118 @@
 # WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 # See the License for the specific language governing permissions and
 # limitations under the License.
-"""Primitives for calling from accelerators to Python functions on the host.
+"""Primitives for calling from JAX accelerator code to Python functions on the host.
 
 **Experimental: please give feedback, and expect changes.**
 
-This module introduces the host callback functions :func:`id_tap` and
-:func:`id_print`, which behave like the identity function but have the
+This module introduces the host callback functions :func:`call`,
+:func:`id_tap`, and :func:`id_print`, that send their arguments from the device
+to the host and invoke user-defined Python functions on the host, optionally
+returning results back to the device computation.
+
+We show below how these functions can be used. We start with :func:`call`,
+and we discuss examples of calling from JAX to NumPy CPU custom kernels,
+or to TensorFlow functions, or to JAX running on another device. In the latter
+two cases we show how we can support JAX autodiff for the host callbacks,
+by deferring to the reverse-mode AD on the target platform. Then we
+show uses of :func:`id_tap` and :func:`id_print`, which have the restriction
+that they cannot return values from the host to the device.
+These primitives are generally faster
+because they are executed asynchronously with the device code and they also
+support the whole spectrum of JAX transformations. In particular, they can be
+used to tap into and to debug JAX-transformed code.
+
+Using :func:`call` to call a host function and return results to device
+-----------------------------------------------------------------------
+
+Use :func:`call` to invoke a computation on the host and return
+NumPy arrays to the device computation.
+Host computation is useful, e.g., when a device computation needs some data
+that requires I/O on the host, or it needs a library that is available on the
+host and you do not want to code it in JAX.
+For example, eigen decomposition in JAX does not work on TPU.
```

Comment:
This is a little misleading -- TPUs support eigen-decomposition for Hermitian matrices only:
```suggestion
For example, eigen decomposition for general matrices in JAX does not work on TPU.
```